app/src/main/java/org/dhis2/data/forms/dataentry/DataEntryAdapter.kt

```diff
@@ -29,7 +33,17 @@ class DataEntryAdapter :
     var sectionPositions: MutableMap<String, Int> = LinkedHashMap()
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FormViewHolder {
-        val layoutInflater = LayoutInflater.from(parent.context)
+        val layoutInflater =
+            if (refactoredViews.contains(viewType) && searchStyle) {
+                LayoutInflater.from(
```

Comment:
I can't inflate from fragment or recycler view, it was not working and style is applied to every item (legacy and refactored). The thing is that with legacy items are crashing (one after another).

1. This will be removed once every view is refactored. 
2. For every new refactored view, it has to be added to the array

If you have any other idea feel free to show it.

Reply:
We'll as we we are working on it, let's iterate. Looks good to me!, good approach.